src/peep/peep.c

```diff
@@ -7772,13 +7772,404 @@ static void peep_on_enter_or_exit_ride(rct_peep *peep, int rideIndex, int flags)
+		peep->photo3_ride_ref = rideIndex;
+
+/// loc_69B36D:
+	if (shopItem == 34)
+		peep->photo4_ride_ref = rideIndex;
+
+/// loc_69B374:
+	peep->window_invalidate_flags |= PEEP_INVALIDATE_PEEP_INVENTORY;
+	peep_update_sprite_type(peep);
+	uint32 ecx = price;
+	uint32 ebx;
+	if (peep->flags & PEEP_FLAGS_TRACKING) {
+		RCT2_GLOBAL(0x13CE952,uint16) = peep->name_string_idx;
+		RCT2_GLOBAL((0x13CE952 + 2), uint32) = peep->id;
+		uint16 bx = shopItem + 0x7F;
+		if (bx >= 0x81) // FIXME: Make sure this comparison is correct
+			bx = 0x5A;
+		// loc_69B3B5:
+		RCT2_GLOBAL((0x13CE956 + 2), uint16) = bx;
+		bx = 0x790;
+		ecx = peep->sprite_index; // This is used later apparently.
+		news_item_add_to_queue(2, bx, ecx);
+	}
+
+/// loc_69B3CC:
+	if (shopItem >= 32) {
+		/// loc_69B455:
+		if (0x00118F48 & (1 << shopItem2))
+			peep->no_of_food++;
+		/// loc_69B465:
+		if (0x00007030 & (1 << shopItem2))
+			peep->no_of_drinks++;
+		/// loc 69B475:
+		if (0x00000087 & (1 << shopItem2))
+			peep->no_of_souvenirs++;
+		/// loc_69B485:
+		// According to the doc on peep_spend_money(), 0x00F1AEC0
+		// is a global var for the "spend type".
+		RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xEA;
+		// Another value that is known, but I don't know
+		// where it's used in the original code.
+		RCT2_GLOBAL(0x0141F56C, uint8) = 0x1C;
+		if (0x00118F48 & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xE8;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		/// loc_69B4B1:
+		if (0x00007030 & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0x46;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		/// loc_69B4CC:
+		ebx = RCT2_ADDRESS(0x00982244, uint16)[shopItem2 * 4]; // TODO: Check stride
+	}
+	else {
+		if (RCT2_GLOBAL(0x12BA3C0, uint32) & (1 << shopItem2))
+			peep->no_of_food++;
+		/// loc_69B3E4:
+		if (0x02400020 & (1 << shopItem2))
+			peep->no_of_drinks++;
+		/// loc_69B3F4:
+		if (0x14001 & (1 << shopItem2))
+			peep->no_of_souvenirs++;
+		/// loc_69B404:
+		RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xEA;
+		RCT2_GLOBAL(0x0141F56C, uint8) = 0x1C;
+		if (RCT2_GLOBAL(0x12BA3C0, uint32) & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0xE8;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		/// loc_69B430:
+		if (0x2400020 & (1 << shopItem2)) {
+			RCT2_GLOBAL(0x00F1AEC0, uint32) = 0x46;
+			RCT2_GLOBAL(0x0141F56C, uint8) = 0x24;
+		}
+		///loc_69B44B:
+		ebx = RCT2_ADDRESS(0x00982164, uint16)[shopItem2 * 4];
+	}
+
+	/// loc_69B4D4:
+	if (RCT2_GLOBAL(RCT2_ADDRESS_PARK_FLAGS, uint32) & 0x800)
+		finance_payment(price, (RCT2_GLOBAL(0x0141F56C, uint8) / 4));
+
+	/// loc_69B4E5:
+	RCT2_GLOBAL(0x0141F56C, uint8) += 0xFC;
+	if (!RCT2_GLOBAL(0x00F1AEE3, unsigned char))
+		peep_spend_money(peep, RCT2_ADDRESS(0x00F1AEC0, money32), price);
+	else {
+		peep->item_standard_flags &= ~PEEP_ITEM_VOUCHER;
+		peep->window_invalidate_flags |= PEEP_INVALIDATE_PEEP_INVENTORY;
+	}
+
+	/// loc_69B50A:
+	ecx -= ebx;
+	RCT2_ADDRESS(0x1362A4C, uint32)[edi] = ecx;
```

Comment:
These are ride struct offsets. This one for example is `ride->total_profit`. You use it slightly differently to the way you have it here. 
```
rct_ride* ride = GET_RIDE(rideIndex); //This does the work of the edi var.
ride->total_profit = ecx;
```
You can tell a ride struct offset as they will be very near the ride_list `0x013628F8` so `0x01362A45 - 0x013628F8 = 0x154` you then look this up in ride.h and see that it is total_profit.